test/python/compiler/test_transpiler.py

```diff
@@ -188,22 +183,21 @@ def test_transpile_non_adjacent_layout(self):
         circuit.cx(qr[1], qr[2])
         circuit.cx(qr[2], qr[3])
 
-        coupling_map = FakeMelbourne().configuration().coupling_map
-        basis_gates = FakeMelbourne().configuration().basis_gates
+        backend = FakeGeneric(num_qubits=6)
         initial_layout = [None, qr[0], qr[1], qr[2], None, qr[3]]
 
         new_circuit = transpile(
             circuit,
-            basis_gates=basis_gates,
-            coupling_map=coupling_map,
+            basis_gates=backend.operation_names,
+            coupling_map=backend.coupling_map,
             initial_layout=initial_layout,
         )
 
         qubit_indices = {bit: idx for idx, bit in enumerate(new_circuit.qubits)}
 
         for instruction in new_circuit.data:
             if isinstance(instruction.operation, CXGate):
-                self.assertIn([qubit_indices[x] for x in instruction.qubits], coupling_map)
+                self.assertIn((qubit_indices[x] for x in instruction.qubits), backend.coupling_map)
```

Comment:
This `assertIn` line doesn't produce a tuple to check, it produces a generator expression which will never appear to be in the coupling map, so if the assertion line triggers it should always fail.  It's not clear to me that this test is doing the right thing at all.